fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/FabricItemSettings.java

```diff
@@ -53,6 +55,44 @@ public FabricItemSettings customDamage(CustomDamageHandler handler) {
 		return this;
 	}
 
+	/**
+	 * Sets the stack-aware recipe remainder provider of the item.
+	 * @see RecipeRemainderProvider
+	 */
+	public FabricItemSettings recipeRemainder(RecipeRemainderProvider provider) {
+		FabricItemInternals.computeExtraData(this).recipeRemainderProvider(provider);
+		return this;
+	}
+
+	/**
+	 * Sets the stack-aware recipe remainder to damage the item by 1 every time it is used in crafting.
+	 */
+	public FabricItemSettings damageIfUsedInCrafting() {
+		return this.damageIfUsedInCrafting(1);
+	}
+
+	/**
+	 * Sets the stack-aware recipe remainder to damage the item by a certain amount every time it is used in crafting.
+	 *
+	 * @param by the amount
+	 */
+	public FabricItemSettings damageIfUsedInCrafting(int by) {
+		return this.recipeRemainder((original, inventory, type, world, pos) -> {
+			if (!original.isDamageable()) {
+				return ItemStack.EMPTY;
+			}
+
+			ItemStack copy = original.copy();
+			copy.damage(by, ThreadLocalRandom.current(), null);
+
+			if (copy.getDamage() >= copy.getMaxDamage()) {
+				return ItemStack.EMPTY;
```

Comment:
What if the item returns a broken value for it (this might be out of the scope for this pr, but its something that might be needed). ie if my hammer becomes a broken hammer or two sticks.

Reply:
should be up to the implementor to do that